WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/Payments Menu/InPersonPaymentsMenu.swift

```diff
@@ -195,8 +195,29 @@ struct InPersonPaymentsMenu: View {
                 },
                                              flow: .creation,
                                              dismissLabel: .backButton,
-                                             viewModel: EditableOrderViewModel(siteID: viewModel.siteID))
+                                             viewModel: viewModel.orderViewModel)
                 .navigationBarHidden(true)
+                .sheet(isPresented: $viewModel.presentCollectPaymentMigrationSheet, onDismiss: {
+                    // Custom amount sheet needs to be presented when the migration sheet is dismissed to avoid conflicting modals.
+                    if viewModel.presentCustomAmountAfterDismissingCollectPaymentMigrationSheet {
+                        viewModel.orderViewModel.addCustomAmount()
+                    }
+                }) {
+                    SimplePaymentsMigrationView {
+                        viewModel.presentCustomAmountAfterDismissingCollectPaymentMigrationSheet = true
+                        viewModel.presentCollectPaymentMigrationSheet = false
+                    }
+                    .presentationDetents([.medium, .large])
+                }
+                .onAppear {
+                    DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
```

Comment:
Is this just for animation purposes when delaying the modal presentation? Or we need to schedule the execution block after 0.5s for other reasons?